src/error.rs

```diff
@@ -0,0 +1,87 @@
+/// Opaque error type for `from_env_ext` function.
+#[derive(Debug)]
+pub struct FromEnvError {
+    pub(crate) inner: FromEnvErrorInner,
+}
+
+/// Public simplified representation of error type for `from_env_ext` function.
```

Comment:
```suggestion
/// Kind of an error returned from `from_env_ext` function.
```
You are writing about implementation details in documentation for users.